packages/debs/amd64/agent/Dockerfile

```diff
@@ -1,35 +1,52 @@
-    apt-get install -y --force-yes \
-    curl gcc make sudo wget expect gnupg perl-base=5.14.2-21+deb7u3 perl \
-    libc-bin=2.13-38+deb7u10 libc6=2.13-38+deb7u10 libc6-dev build-essential \
-    cdbs devscripts equivs automake autoconf libtool libaudit-dev selinux-basics \
-    libdb5.1=5.1.29-5 libdb5.1-dev libssl1.0.0=1.0.1e-2+deb7u20 procps gawk libsigsegv2
-
-RUN apt-get update && apt-get build-dep python3.2 -y --force-yes
-
-RUN curl -OL http://packages.wazuh.com/utils/gcc/gcc-9.4.0.tar.gz && \
-    tar xzf gcc-9.4.0.tar.gz  && cd gcc-9.4.0/ && \
+RUN apt-get update && apt-get install -y build-essential \
+            curl wget git gawk devscripts sudo libssl-dev \
+            unzip zip tar texinfo pkg-config clang-tidy uuid \
+            libmpfr-dev libgmp3-dev libmpc-dev autopoint libtool \
+            zlib1g-dev libgcrypt20-dev libmagic-dev libpopt-dev \
+            libmagic-dev libsqlite3-dev  gettext \
+            ninja-build libsystemd-dev libaudit-dev selinux-basics \
+            libarchive-dev debhelper libssl-dev libbz2-dev \
+            && rm -rf /var/lib/apt/lists/*
+
+RUN echo "deb http://ftp.de.debian.org/debian bullseye main" >>/etc/apt/sources.list && \
+    apt-get update && apt-get install -y g++-10 && rm -rf /var/lib/apt/lists/*
+
+RUN wget https://ftp.gnu.org/gnu/gcc/gcc-13.2.0/gcc-13.2.0.tar.gz && \
+    tar xzf gcc-13.2.0.tar.gz  && cd gcc-13.2.0/ && \
     ./contrib/download_prerequisites && \
-    ./configure --prefix=/usr/local/gcc-9.4.0 --enable-languages=c,c++ --disable-multilib \
-        --disable-libsanitizer && \
-    make -j$(nproc) && make install && \
-    ln -fs /usr/local/gcc-9.4.0/bin/g++ /usr/bin/c++ && \
-    ln -fs /usr/local/gcc-9.4.0/bin/gcc /usr/bin/cc && cd / && rm -rf gcc-*
-
-ENV CPLUS_INCLUDE_PATH "/usr/local/gcc-9.4.0/include/c++/9.4.0/"
-ENV LD_LIBRARY_PATH "/usr/local/gcc-9.4.0/lib64:${LD_LIBRARY_PATH}"
-ENV PATH "/usr/local/gcc-9.4.0/bin:${PATH}"
-
-RUN curl -OL http://packages.wazuh.com/utils/cmake/cmake-3.18.3.tar.gz && \
-    tar -zxf cmake-3.18.3.tar.gz && cd cmake-3.18.3 && \
-    ./bootstrap --no-system-curl CXX=/usr/local/gcc-9.4.0/bin/g++ \
-        CC=/usr/local/gcc-9.4.0/bin/gcc && \
+    ./configure --prefix=/usr/local/gcc-13.2.0 --enable-languages=c,c++ \
+        --disable-multilib --disable-libsanitizer && \
+    make -j$(nproc) && make install -j$(nproc) && \
+    ln -fs /usr/local/gcc-13.2.0/bin/g++ /usr/bin/c++ && \
+    ln -fs /usr/local/gcc-13.2.0/bin/gcc /usr/bin/cc && cd .. && rm -rf gcc-*
+
+ENV PKG_CONFIG_PATH "/usr/local/lib/pkgconfig:${PKG_CONFIG_PATH}"
+
+RUN curl -OL https://github.com/Kitware/CMake/releases/download/v3.30.3/cmake-3.30.3.tar.gz && \
+    tar -zxf cmake-3.*.tar.gz && cd cmake-3.* && ./bootstrap && \
     make -j$(nproc) && make install && ln -s /usr/local/bin/cmake /usr/bin/cmake && \
     cd / && rm -rf cmake-*
+
+RUN curl -sO https://lua.org/ftp/lua-5.4.7.tar.gz && tar -xzvf lua-5.4.7.tar.gz && \
```

Comment:
Is installing Lua necessary? I believed that Lua as a dependency was introduced in one of the later dependencies upgrades in version 4, but then there was no `vcpkg`, and now Lua is included in `vcpkg.json`.

Reply:
This is necessary, because without this dependency previously installed, compiling the overlay port in `vcpkg` fails.